pkg/cars/kubectl/main.go

```diff
@@ -0,0 +1,134 @@
+package kubectl
+
+import (
+	"github.com/jtyr/gbt/pkg/core/car"
+	"github.com/jtyr/gbt/pkg/core/utils"
+	"regexp"
+	"os/exec"
+)
+
+// Car inherits the core.Car.
+type Car struct {
+	car.Car
+}
+
+type kubeContextInfo struct {
+	context   string
+	cluster   string
+	authInfo  string
+	namespace string
+}
+
+var kubectlLookupCmd = "kubectl"
+var runGetContexts = []string{"kubectl", "config", "get-contexts"}
+
+// Returns true if kubectl command exists
+func isKubectlExists() bool {
+	_, err := exec.LookPath(kubectlLookupCmd)
+
+	return err == nil
+}
+
+// return the current context information for kubectl
+func getCurrentContext(display bool) *kubeContextInfo {
+	if !display {
+		return &kubeContextInfo{
+			namespace: "",
+			context:   "",
+			cluster:   "",
+			authInfo:  "",
+		}
+	}
+
+	rc, out, _ := utils.Run(runGetContexts)
+
+	matchGroupMapping := make(map[string]string)
+	if rc == 0 {
+		var re = regexp.MustCompile(`(?m)^\*[\t\f\v ]+(?P<Context>[\w\-]+)[\t\f\v ]+(?P<Cluster>[\w\-]+)\s+(?P<AuthInfo>[\w\-]+)(?:[\t\f\v ]+(?P<Namespace>[\w\-]+)?)?$`)
+		groupNames := re.SubexpNames()
+		for _, match := range re.FindAllStringSubmatch(out, -1) {
+			for groupIdx, value := range match {
+				name := groupNames[groupIdx]
+				if name == "" {
+					name = "*"
+				}
+
+				matchGroupMapping[name] = value
+			}
+		}
+	}
+
+	return &kubeContextInfo{
+		cluster:   matchGroupMapping["Cluster"],
+		context:   matchGroupMapping["Context"],
+		authInfo:  matchGroupMapping["AuthInfo"],
+		namespace: matchGroupMapping["Namespace"],
+	}
+}
+
+// Init initializes the car.
+func (c *Car) Init() {
+	defaultRootBg := utils.GetEnv("GBT_CAR_BG", "light_blue")
+	defaultRootFg := utils.GetEnv("GBT_CAR_FG", "white")
+	defaultRootFm := utils.GetEnv("GBT_CAR_FM", "none")
+	defaultIconBg := defaultRootBg
+	defaultIconFg := defaultRootFg
+	defaultIconFm := defaultRootFm
+	defaultContextBg := defaultRootBg
+	defaultContextFg := defaultRootFg
+	defaultContextFm := defaultRootFm
+	defaultClusterBg := defaultRootBg
+	defaultClusterFg := defaultRootFm
+	defaultClusterFm := defaultRootFm
+	defaultAuthInfoBg := defaultRootBg
+	defaultAuthInfoFg := defaultRootFm
+	defaultAuthInfoFm := defaultRootFm
+	defaultNamespaceBg := defaultRootBg
+	defaultNamespaceFg := defaultRootFm
+	defaultNamespaceFm := defaultRootFm
+
+	c.Display = utils.GetEnvBool("GBT_CAR_KUBECTL_DISPLAY", isKubectlExists())
+	contextInfo := getCurrentContext(c.Display)
+
+	c.Model = map[string]car.ModelElement{
+		"root": {
+			Bg:   utils.GetEnv("GBT_CAR_KUBECTL_BG", defaultRootBg),
```

Comment:
Please could you format it like in the other cars? Also we do not use tabs but 4 spaces. I know, it's not the recommended way, but I believe it improves code readability.

Reply:
> Please could you format it like in the other cars?

I don't follow. Do you mean the way the lines are broken up?

> Also we do not use tabs but 4 spaces. I know, it's not the recommended way, but I believe it improves code readability.

My bad, IntelliJ has been set to use tabs by default for go for some reason, had forgotten to change it to spaces.